test/data-factory.js

```diff
@@ -1,3 +1,5 @@
+import { date } from "zod";
+import { createDuty } from "../src/models/duty";
 import { getSoldierRank } from "../src/models/soldier";
 
 function generatePostSoldier(soldierParams) {
```

Comment:
```suggestion
function generatePostSoldier(soldierParams = {}) {
```

default to empty object in all generators, so you don't need to explicitly pass an empty object in your tests